contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,550 @@
+* [Non-Goals](#non-goals)
+* [Current State of Things](#current-state-of-things)
+* [CSPC Schema Proposal](#cspc-schema-proposal)
+
+## Introduction
+
+This proposal highlights the limitations of current CSPC and CSPI schema and
+proposes improvements for the same.
+Please refer to the following link for cstor operator design document. 
+https://github.com/openebs/openebs/blob/master/contribute/design/1.x/cstor-operator/doc.md
+
+# Goals
+
+The major goal of this document is to freeze the schema for CSPC and CSPI schema.
+Apart from this the document focuses on following aspects too:
+
+- Migrating the CSPC and CSPI CRD to v1 apiversion.
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level statuses fields on CSPC/CSPI.
+
+# Non-Goals
+
+- Introduction of new fields in the CSPC/CSPI to deliver feature etc.
+- List of possible events that should be recorded on CSPC and CSPI.
+- List of conditions that should be put on the CSPC and CSPI CRs.
+
+**Note:** Conditions vs Events
+
+Events can be of "Normal" or "Warning" only as of now in Kubernetes and components that processes the API object
+(in this case, the CSPC or CSPI CR) can record some useful event on the custom resource(CR). Events let us understand the state of action performed by the component for the object (CR) and if those are recorded to the CR object, it can help in debugging. 
+One typical example is -- Your pod was not created as part of applying the deployment object or the pods are in non-running state then we can describe the deployment object and see the events recorded by different components e.g. scheduler, attach-detach controller to understand why such thing happened.
+
+Conditions can be a set of well-defined state that can occur for an API object.
+For example, POD API object has "PodScheduled" as a conditions ( value being true of false ) and status.phase field in the API object that gives an high level overview of the pod e.g. Running, Pending etc.
+ 
+
+## Current State of Things
+
+### CSPC Schema
+
+The CSPC API has the following capabilities : 
+ 
+- CSPC API can be used to provision cStor pools on a single or multiple nodes. Pool configuration for each node can be specified in the CSPC. 
+
+- Stripe, mirror, raidz1, and raidz2 are the only supported raid topologies.
+
+
+- A single cStor pool on a node can have any number of raid groups. API schema can support heterogeneous raid topologies. ( e.g One raid group of mirror other of raidz and so on). ( Not sure if it works from the data plane side and can be of niche or trivial use case )
+
+
+- A stripe raid group can have any number of block devices but not less than 1.
+
+
+- A mirror, raidz and raidz2 raid group can only have exactly 2, 3 and 6 block devices only.  
+
+
+- A raid group can be designated to be a write cache, read cache or spare, apart from regular vdev in a pool.
+
+
+- CSPC has the capability to specify a cache file for faster imports.
```

Comment:
what does import mean here?

Reply:
When the cStor pool is created with `cache file` option then disk configuration will be stored in cache file(on which disks pool is configured). Whenever the cStorpool container is restarted then the pool needs to be imported to perform operations. During pool import, we can specify the cachefile which will be importing the pool directly from those pool disks. If cache file is not specified then zrepl(process inside the pool container) will look into all the disks attached to the node and import the matching pool.

import means bringing pool existence.